### ISSUE TITLE
Nested divs lead to jumping cursor on insert

### ISSUE THREAD

**Issue by dandormont** [quilljs/quill#484]
Sorry, can't find a better way to explain this but I made a JSFiddle example here: http://jsfiddle.net/fjqbL1tw/1/

Run that example, and click into the middle of the bolded "Line 2". Then, type a letter. Notice that the caret jumps to the very beginning of the document. Happens in Chrome 45 and Firefox 39 with QuillJS 0.20.0

I was able to fix it locally by making a change similar to this in the "findLine" method (this is the JS version, sorry I don't have a good CoffeeScript setup here):

    while ((node != null) && (dom.BLOCK_TAGS[node.tagName] == null || dom(node).data(Line.DATA_KEY) == null)) {
 
In other words, it attempts to handle a div that isn't necessarily the line marker by continuing to look up the tree. But I'm not sure this is the right way to go, or what side effects it might have.

**Comment by jhchen** [comment]
This HTML markup is invalid. You can't have a block level tag like a `<div>` inside of an inline one like a `<b>`.

**Comment by dandormont** [comment]
Well, a user of mine produced content with exactly that combination of elements and as far as I can tell, they did it using Quill. I can't prove that with an exact stream of inputs, but it doesn't seem terribly likely that they manipulated the DOM on their own somehow, though I will investigate further and open a new issue if I find anything.
